hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/PrisonApiService.kt

```diff
@@ -0,0 +1,57 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.search.services
+
+import kotlinx.coroutines.async
+import kotlinx.coroutines.reactive.awaitSingle
+import kotlinx.coroutines.runBlocking
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.bodyToMono
+
+@Component
+class PrisonApiService(val prisonApiWebClient: WebClient) {
+
+  fun getAllAlerts(): List<AlertType> {
+    return runBlocking {
+      val alertTypes = async { getDomainReferenceCodes("ALERT") }
```

Comment:
I should have mentioned that there is an endpoint in Prison API to return the nested data - https://prison-api-dev.prison.service.justice.gov.uk/swagger-ui/index.html#/reference-domains/getAlertTypes rather than having to get the reference domains separately